pkg/deployment/resources/servicemonitor.go

```diff
@@ -0,0 +1,118 @@
+//
+// DISCLAIMER
+//
+// Copyright 2019 ArangoDB Inc, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Max Neunhoeffer
+//
+
+package resources
+
+import (
+	"github.com/arangodb/kube-arangodb/pkg/util/k8sutil"
+
+	coreosv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	clientv1 "github.com/coreos/prometheus-operator/pkg/client/versioned/typed/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+func LabelsForExporterServiceMonitor(deploymentName string) map[string]string {
+	return map[string]string{
+		k8sutil.LabelKeyArangoDeployment: deploymentName,
+		k8sutil.LabelKeyApp:              "arango-exporter",
+		"context":                        "metrics",
```

Comment:
Dislike that but if ewout insists ...

Reply:
Need to check with a stock Prometheus deployment, will do.

Reply:
Any update on this?

Reply:
There is no "generic" Prometheus deployment. But I have used the stock prometheus-operator deployment from their web page and added a single line for the serviceMonitorSelector and it worked. I have added an additional label `metrics: prometheus` and documented this. I think this is enough.